app/src/pages/contact.tsx

```diff
@@ -1,76 +1,74 @@
-import { useAppContext } from '@context/state'
-import type {
-  GetServerSideProps,
-  GetServerSidePropsResult,
-  NextPage,
-} from 'next'
+import type { GetServerSideProps, NextPage } from 'next'
 import { Trans } from 'next-i18next'
 import { serverSideTranslations } from 'next-i18next/serverSideTranslations'
-import { ChangeEventHandler, useEffect, useState } from 'react'
-import NumberFormat from 'react-number-format'
+import { ChangeEvent, useEffect, useState } from 'react'
+import { PatternFormat } from 'react-number-format'
 
 import Alert from '@components/Alert'
 import BackLink from '@components/BackLink'
 import ButtonLink from '@components/ButtonLink'
 import Required from '@components/Required'
 import RequiredQuestionStatement from '@components/RequiredQuestionStatement'
-import TextArea from '@components/TextArea'
-import TextInput from '@components/TextInput'
+import TextField from '@components/TextField'
 
-interface Props {
-  previousRoute: string
-}
-
-const Contact: NextPage<Props> = (props: Props) => {
-  const { session, setSession } = useAppContext()
-  const [form, setForm] = useState(session?.contact)
-  const [continueBtn, setContinueBtn] = useState<{
-    labelKey: string
-  }>({ labelKey: 'continue' })
-  const requiredMet = (): boolean => {
-    const validPhoneLength = form.phone.replace(/[^0-9]/g, '').length === 10
-    return (
-      validPhoneLength &&
-      ['firstName', 'lastName', 'phone'].every(
-        (field) => form[field as keyof typeof form]
-      )
-    )
-  }
-  const [disabled, setDisabled] = useState<boolean>(!requiredMet())
+import type { ContactData, EditablePage } from '@src/types'
+import { isValidContact } from '@utils/dataValidation'
+import { initialContactData } from '@utils/sessionData'
 
+const Contact: NextPage<EditablePage> = (props: EditablePage) => {
+  // Get the session from props.
+  const {
+    session,
+    setSession,
+    reviewMode = false,
+    backRoute = '',
+    forwardRoute = '',
+  } = props
+  // Initialize form as a state using the value with default blank values.
+  // This prevents bugginess around hydration. DO NOT try to initialize from the session!
+  // That can cause fields like <textarea> to not load values correctly.
+  const [form, setForm] = useState<ContactData>(initialContactData)
+  // Need to use useEffect() to properly load the data from session storage on component mount.
+  // This also updates form whenever session is updated, which isn't ideal since
+  // we update form AND session in handleChange(), but:
+  // 1. React requires session.contact as a dependency
+  // 2. We could remove setForm() from handleChange(), but that would put the update first
+  //    into sessionStorage before updating state and that is less ideal.
+  // Instead, we accept that this runs on component mount and on handleChange(), we run setForm(),
+  // then, setSession(), which then calls setForm() a second time.
```

Comment:
Notes for reviewers: I discussed this a little bit in the comment https://github.com/navapbc/wic-mt-demo-project-eligibility-screener/pull/146/commits/5ecfb3e04ec3ae437e6312094eaa54bf2c284109#r1007830593. 

**Question:** Is there a better way to handle this? Do you disagree with this pattern?